h2/src/main/org/h2/mvstore/Page.java

```diff
@@ -575,24 +563,20 @@ public void insertLeaf(int index, Object key, Object value) {
      * @param key the key
      * @param childPage the child page
      */
-    public void insertNode(int index, Object key, Page childPage) {
-
-        Object[] newKeys = new Object[keys.length + 1];
-        DataUtils.copyWithGap(keys, newKeys, keys.length, index);
-        newKeys[index] = key;
+    public abstract void insertNode(int index, Object key, Page childPage);
+
+    @SuppressWarnings("SuspiciousSystemArraycopy")
```

Comment:
There is no `System.arraycopy()` inside this method.

Reply:
Not yet, back-port hiccup